driver/csiplugin/controllerserver.go

```diff
@@ -1223,27 +1258,7 @@ func (cs *ScaleControllerServer) DeleteVolume(ctx context.Context, req *csi.Dele
 				}
 
 				if volumeIdMembers.StorageClassType == STORAGECLASS_ADVANCED {
-					glog.V(4).Infof("trying to delete independent fileset for consistency group [%v]", volumeIdMembers.ConsistencyGroup)
-					filesets, err := conn.GetFilesetsInodeSpace(FilesystemName, inodeSpace)
-					if err == nil {
-						found := false
-						if len(filesets) > 1 {
-							found = true
-							glog.V(4).Infof("found atleast one dependent fileset for consistency group: [%v]", volumeIdMembers.ConsistencyGroup)
-						}
-
-						if !found {
-							// Delete independent fileset for consistency group
-							_, err := cs.DeleteFilesetVol(FilesystemName, volumeIdMembers.ConsistencyGroup, volumeIdMembers, conn)
-							if err != nil {
-								glog.V(4).Infof("deletion of independent fileset for consistency group [%v] failed with error: [%v]", volumeIdMembers.ConsistencyGroup, err)
-							} else {
-								glog.V(4).Infof("deleted independent fileset for consistency group [%v]", volumeIdMembers.ConsistencyGroup)
-							}
-						}
-					} else {
-						glog.V(4).Infof("listing of filesets for filesystem: [%v] failed with error: [%v]", FilesystemName, err)
-					}
+					cs.DeleteCGFileset(FilesystemName, inodeSpace, volumeIdMembers, conn)
```

Comment:
Shouldn't we be checking error if deletion of independent fileset failed ?

Reply:
Currently one reason to not handle error is that even if we fail we want to continue in this function and delete symbolic link. Another question is here we have already successfully deleted the dependent fileset for the volume and what has failed is deletion of independent fileset. If we return error here then kubernetes will keep on retrying, do we want to have that behavior ?

Reply:
lets consider the this is last volume from CG and for some reason GUI misbehaved during deletion of independent fileset deletion and independent fileset remained there then who is going to delete it ? 

if volume deletion fails in independent fileset and csi returns error, k8s going retry again and again until all the required steps are done from deletion perspective. We should consider deletion of independent fileset as one individual step and should be retried as part of idempotency.

Reply:
Done